### ISSUE TITLE
.vtx strips' has flags equal to "\u0001"

### ISSUE THREAD

**Issue by AndreasWJ** [LordVonAdel/source-mdl#3]
Hi! First off, hats off to all collaborators on this module, it's excellent!
Secondly, I've been running into issues importing, then exporting as GLTF, and I hope someone could point me in the right direction. The error became evident as I imported the exported GLTF file into 3D software, where I receive the following errors:
```
Error	Message	Pointer
VALUE_NOT_IN_RANGE	Value 0 is out of range.	/accessors/3/count
VALUE_NOT_IN_RANGE	Value 0 is out of range.	/bufferViews/3/byteLength
BUFFER_VIEW_TOO_LONG	BufferView does not fit buffer (0) byteLength (400736).	/bufferViews/3/byteOffset
```

Here's my code, I've simplified it enough to rule out interference from business logic etc:
```
const model = new MDL();
fs.readFile('data/extracted/models/weapons/v_rif_ak47.mdl', (err, mdlData) => {
	fs.readFile('data/extracted/models/weapons/v_rif_ak47.dx90.vtx', (err, vtxData) => {
		fs.readFile('data/extracted/models/weapons/v_rif_ak47.vvd', (err, vvdData) => {
			console.log(mdlData.length, vvdData.length, vtxData.length);
			console.log(mdlData.byteLength, vvdData.byteLength, vtxData.byteLength);
			// Import the three buffers
			model.import({ mdlData, vtxData, vvdData });

			// If you want, look at some metadata
			console.log(model.getMetadata());

			// Export as GLTF
			fs.writeFileSync('data/turret.gltf', JSON.stringify(model.toGLTF()));
		});
	});
});
```
v_rif_ak47.mdl, v_rif_ak47.dx90.vtx, v_rif_ak47.vvd, are all unpackaged from CSGO's vpk archives and are in their correct directories.

----------------------------------------------------------------------------------------------
From me investigating the cause I found out that my Source files are imported and exported correctly through this module's demo(the webpage). Yet, it fails the same process in the rudimentary code snippet above. I then compared output's from the demo's webpage and from my NodeJS app and found that the fourth accessor had a count of 0, which originates from:
toGLTF.js: 94-100
```
accessors.push({
      bufferView: 3,
      componentType: 5123,
      count: mesh.indices.length,
      byteOffset: meshOffsets[i],
      type: "SCALAR"
    });
```

`mesh.indices.length` should not equal 0 to my understanding. I further read into Import.js where mesh.indices is defined, particularly the code snippet below:
Import.js: 109-129
```
for (let stripGroup of meshVTX.stripGroups) {

              for (let strip of stripGroup.strips) {
                if (strip.flags & STRIP_IS_TRILIST) {
                  for (let i = strip.numIndices - 1; i >= 0; i--) {
                    let vtxId = stripGroup.indices[i + strip.indexOffset];
                    let vtx = stripGroup.vertices[vtxId];
                    mesh.indices.push(vtx.origMeshVertID + meshMDL.vertexoffset);
                  }
                } else if (strip.flags & STRIP_IS_TRISTRIP) {
                  /**
                   * @todo Test it with a model that uses tristrips
                   */
                  for (let i = strip.numIndices + strip.indexOffset; i >= strip.indexOffset + 2; i--) {
                    mesh.indices.push(
                      stripGroup.vertices[stripGroup.indices[i]].origMeshVertID + meshMDL.vertexoffset,
                      stripGroup.vertices[stripGroup.indices[i - 2]].origMeshVertID + meshMDL.vertexoffset,
                      stripGroup.vertices[stripGroup.indices[i - 1]].origMeshVertID + meshMDL.vertexoffset
                    );
                  }
                }
              }
```

For some reason, on my NodeJS version etc etc, strip.flags equals the string "\u0001", therefore not entering the if block:
`if (strip.flags & STRIP_IS_TRILIST)`. From my limited knowledge, "\u0001", is a unicode literal. Meanwhile, the demo webpage evaluates strip.flags to an integer, 1, therefore it enters the if block as expected.

Any help means a lot! Sorry for the long text, wanted to provide some context.

----------------------------------------------------------------------------------------------

source-mdl version: 0.2.0
npm version: 6.14.10
node version: v14.15.3

**Comment by AndreasWJ** [comment]
Found out the cause!
In one of the dependencies, struction/types.js:143:
```
// Will be now (package version 0.3.0) converted to string. Hope this will not break anything
      return String.fromCharCode(buffer.readUInt8(offset));
```

Reading strip.flags returns an unknown ASCII character for me. If I would remove the conversion to ASCII in structron or convert it back to an integer via `charCodeAt(...)` it should work as expected! "\u0001" was returned because I had wrapped it in a JSON.stringify(...).

@LordVonAdel Does the demo webpage have an old version of structron without the ASCII conversion? It would explain the difference between my NodeJS and the demo webpage's.
